pkg/api/admin/openshiftcluster_convert.go

```diff
@@ -292,13 +295,16 @@ func (c openShiftClusterConverter) ToInternal(_oc interface{}, out *api.OpenShif
 	}
 	if oc.Properties.PlatformWorkloadIdentityProfile != nil && oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities != nil {
 		out.Properties.PlatformWorkloadIdentityProfile = &api.PlatformWorkloadIdentityProfile{}
-		out.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities = make([]api.PlatformWorkloadIdentity, len(oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities))
+		out.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities = make(map[string]api.PlatformWorkloadIdentity, len(oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities))
 
-		for i := range oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities {
-			out.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities[i].OperatorName = oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities[i].OperatorName
-			out.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities[i].ResourceID = oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities[i].ResourceID
-			out.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities[i].ClientID = oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities[i].ClientID
-			out.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities[i].ObjectID = oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities[i].ObjectID
+		for k := range oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities {
```

Comment:
This logic should be able to be simplified similar to how the tags are formed, isn't it?  
```
for k, v := range oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities {
    out.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities[k].ClientId == v.ClientId
...

Reply:
I tried that first, but was unable to get it to compile:
```
cannot assign to struct field out.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities[k].ResourceID in map
```
etc...